src/highlight.rs

```diff
@@ -1317,7 +1315,7 @@ impl<'s> Highlighter<'s> {
             self.pending_variables.push(var_name);
         }
     }
-    fn visit_semi_nl(&mut self, node: &dyn Node) {
+    fn visit_semi_nl(&mut self, node: NodeRef<'_>) {
```

Comment:
`visit_semi_nl` can now take a `semi_nl: &SemiNl`.
Since `color_node` takes `impl Node`, this will work fine.

Same for other `visit_semi_nl` functions.

Reply:
Done.